share/completions/magento.fish

```diff
@@ -15,13 +15,17 @@ end
 ###
 
 function __fish_print_magento_modules -d "Lists all Magento modules"
-    set -l modules (magento module:status)
-
-    for i in $test
-        if test -n "$i" -a "$i" != None
-            echo $i
-        end
-    end
+    test -f app/etc/config.php; or return
+    command -q php; or return
+
+    php -r '
+        $config  = include "app/etc/config.php";
```

Comment:
I'm not sure i understand what you mean. Can you elaborate?

Reply:
`include` is php's equivalent of `source` in shell. It *runs* the given file and keeps the definitions around.

That means, if your current directory includes a file in "app/etc/config.php", completing `magento` would run that entire file, no matter what code it contained. It could do *anything*.

And the name doesn't mention "magento" anywhere, it's conceivable that a file at that place would exist and not be meant for magento at all.

Reply:
Ah, I get your point. Yes that that is a security threat indeed. In Magento projects the file `app/etc/config.php` is used to define multiple things that are necessary for Magento to run, amongst others the list of active and inactive modules. Usually the file contains a huge array that is being returned, e.g.:
```php
return [
    'modules' => [
        'Module_Name1' => 1,
        'InactiveMModule_Name1' => 0,
        ...
    ],
    'config' => [
        ...
    ]
];
```

Magento itself includes the files in the same manner as I did in the code of my PR. Yet there is always a risk, that someone has added malicious code to the file, as you have elaborated. This would result in an infected Magento installation as the same file is being used in production, which would bring the thing to a completely different level.
Yet, you are right in feeling bad about this as it could be an attack targeted at `fish` only. Would it be an option to execute the code in a subshell? If not, I'd revert the code to execute `bin/magento` and accept that it is less performant.

Reply:
> Would it be an option to execute the code in a subshell?

The code is unable to change anything inside fish. It runs in a php process.

But that php process can do arbitrary things, including change any file or run any other command.

The *shell* isn't the problem, so a subshell wouldn't help.

>  I'd revert the code to execute bin/magento and accept that it is less performant

That would be preferable, but still not ideal. Really all it would change is that we could blame magento for using a script file (instead of e.g. a json file) as its config.

So I'm not sure if we want to include this at all and shouldn't just accept that we can't complete magento without running arbitrary code.

Reply:
> That would be preferable, but still not ideal. Really all it would change is that we could blame magento for using a script file (instead of e.g. a json file) as its config.

True yet that is the state it was in before I fixed the error. Of course this whole part of the completion could be dumped but considering that usually a lot of modules are being used by Magento, this part of the completion is quite comfortable to have. What would you suggest?

Reply:
Having thought about this some more, I believe this part of the completion needs to be removed. Anyone who wants it can get it elsewhere, but fish completions executing random scripts in $PWD is surprising and a potential problem, especially considering how we changed the git prompt to avoid such things.

Sorry.

If you want, you can change it here, or I'll close this and do it myself.

Reply:
I reimplemented said completion using a `sed` approach. Better?